mat64/dense_arithmetic.go

```diff
@@ -709,7 +709,9 @@ func (m *Dense) Pow(a Matrix, n int) {
 		if n&1 != 0 {
 			w.Mul(&w, &tmp)
 		}
-		tmp.Mul(&tmp, &tmp)
```

Comment:
Having said that, there is another optimisation here that I will try later today.